### PR TITLE
Moving DSC common classes into service management library, also skipp…

### PR DESCRIPTION
**Pull request by markcowl**
…ing failing Sql Test

### CODE REVIEW

**Comment by markcowl** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/123/

**Comment by markcowl** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/420/

File: src/Common/Commands.Common.Test/Commands.Common.Test.csproj [hunk 1]
```diff
@@ -314,4 +311,4 @@
     </PropertyGroup>
     <Error Condition="!Exists('..\..\packages\xunit.runner.visualstudio.2.0.0\build\net20\xunit.runner.visualstudio.props')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\xunit.runner.visualstudio.2.0.0\build\net20\xunit.runner.visualstudio.props'))" />
```
Comment (eshaparmar) [inline src/Common/Commands.Common.Test/Commands.Common.Test.csproj]: There are some resource files - under Commands.Common.Test\Resources\DSC - that needs to be moved to ServiceManagement.Test project.

File: src/Common/Commands.Common/Commands.Common.csproj [hunk 2]
```diff
@@ -157,14 +141,6 @@
     <Compile Include="AzurePowerShell.cs" />
     <Compile Include="ComputeCloudException.cs" />
     <Compile Include="Constants.cs" />
-    <Compile Include="Extensions\DSC\DscExtensionCmdletConstants.cs" />
```
Comment (eshaparmar) [inline src/Common/Commands.Common/Commands.Common.csproj]: These common classes are referencing some strings from the Commands.Common\Properties\Resources.resx file
These one in particular. Need to move these to the Commands.Compute\Properties\Resources.resx file and ServiceManagement\Properties\Resources.resx file

<data name="PublishVMDscExtensionGetDscResourceFailed" xml:space="preserve">
    <value>Cannot get module for DscResource '{0}'. Possible solutions:
1) Specify -ModuleName for Import-DscResource in your configuration.
2) Unblock module that contains resource.
3) Move Import-DscResource inside Node block.
</value>
    <comment>0 = name of DscResource</comment>
  </data>
  <data name="PublishVMDscExtensionRequiredPsVersion" xml:space="preserve">
    <value>Your current PowerShell version {1} is less then required by this cmdlet {0}. Consider download and install latest PowerShell version.</value>
    <comment>{0} = minimal required PS version, {1} = current PS version</comment>
  </data>
  <data name="AzureVMDscParsingConfiguration" xml:space="preserve">
    <value>Parsing configuration script: {0}</value>
    <comment>{0} is the path to a script file</comment>
  </data>
  <data name="PublishVMDscExtensionStorageParserErrors" xml:space="preserve">
    <value>Configuration script '{0}' contained parse errors:
{1}</value>
    <comment>0 = path to the configuration script, 1 = parser errors</comment>
  </data>
  <data name="PublishVMDscExtensionRequiredModulesVerbose" xml:space="preserve">
    <value>List of required modules: [{0}].</value>
    <comment>{0} = list of modules</comment>
  </data>
  <data name="PublishVMDscExtensionTempFolderVerbose" xml:space="preserve">
    <value>Temp folder '{0}' created.</value>
    <comment>{0} = temp folder path</comment>
  </data>
  <data name="PublishVMDscExtensionCopyFileVerbose" xml:space="preserve">
    <value>Copy '{0}' to '{1}'.</value>
    <comment>{0} = source, {1} = destination</comment>
  </data>
  <data name="PublishVMDscExtensionCopyModuleVerbose" xml:space="preserve">
    <value>Copy the module '{0}' to '{1}'.</value>
    <comment>{0} = source, {1} = destination</comment>
  </data>
  <data name="AzureVMDscArchiveAlreadyExists" xml:space="preserve">
    <value>File '{0}' already exists. Use the -Force parameter to overwrite it.</value>
    <comment>{0} is the path to a file</comment>
  </data>
  <data name="PublishVMDscExtensionUploadArchiveConfigFileNotExist" xml:space="preserve">
    <value>Configuration file '{0}' not found.</value>
    <comment>0 = path to the configuration file</comment>
  </data>
  <data name="PublishVMDscExtensionDirectoryNotExist" xml:space="preserve">
    <value>Path '{0}' not found.</value>
    <comment>0 = path to the additional content file/directory</comment>
  </data>
  <data name="PublishVMDscExtensionAdditionalContentPathNotExist" xml:space="preserve">
    <value>Path '{0}' not found.</value>
    <comment>0 = path to the additional content file/directory</comment>
  </data>
  <data name="PublishVMDscExtensionUploadArchiveConfigFileInvalidExtension" xml:space="preserve">
    <value>Invalid configuration file: {0}.
The file needs to be a PowerShell script (.ps1 or .psm1) or a ZIP archive (.zip).</value>
    <comment>0 = path to the configuration file</comment>
  </data>
  <data name="PublishVMDscExtensionCreateArchiveConfigFileInvalidExtension" xml:space="preserve">
    <value>Invalid configuration file: {0}.
The file needs to be a PowerShell script (.ps1 or .psm1).</value>
    <comment>0 = path to the configuration file</comment>
  </data>
  <data name="AzureVMDscCreateArchiveAction" xml:space="preserve">
    <value>Create Archive</value>
  </data>
  <data name="AzureVMDscUploadToBlobStorageAction" xml:space="preserve">
    <value>Upload '{0}'</value>
    <comment>{0} is the name of an storage blob</comment>
  </data>
  <data name="AzureVMDscStorageBlobAlreadyExists" xml:space="preserve">
    <value>Storage Blob '{0}' already exists. Use the -Force parameter to overwrite it.</value>
    <comment>{0} is the name of an storage blob</comment>
  </data>
  <data name="PublishVMDscExtensionArchiveUploadedMessage" xml:space="preserve">
    <value>Configuration published to {0}</value>
    <comment>{0} is an URI</comment>
  </data>
  <data name="PublishVMDscExtensionDeletedFileMessage" xml:space="preserve">
    <value>Deleted '{0}'</value>
    <comment>{0} is the path of a file</comment>
  </data>
  <data name="PublishVMDscExtensionDeleteErrorMessage" xml:space="preserve">
    <value>Cannot delete '{0}': {1}</value>
    <comment>{0} is the path of a file, {1} is an error message</comment>
  </data>
<data name="AzureVMDscCannotFindConfigurationDataFile" xml:space="preserve">
    <value>Cannot find configuration data file: {0}</value>
  </data>
  <data name="AzureVMDscInvalidConfigurationDataFile" xml:space="preserve">
    <value>The configuration data must be a .psd1 file</value>
  </data>

Comment (markcowl) [inline src/Common/Commands.Common/Commands.Common.csproj]: Both of these projects include commands.common, so this is not actually necessary - many commands put their resources in that resources file.

Comment (eshaparmar) [inline src/Common/Commands.Common/Commands.Common.csproj]: OK.

File: src/Common/Commands.Common.Test/Commands.Common.Test.csproj (hunk 1 above)
Comment (eshaparmar) [inline src/Common/Commands.Common.Test/Commands.Common.Test.csproj]: As discussed offline...this can be fixed later. Pls ignore.

**Comment by markcowl** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/130/

**Comment by markcowl** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/425/

**Pull request merged**